targets/TARGET_RDA/TARGET_UNO_91H/sleep.c

```diff
@@ -0,0 +1,58 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2013 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "sleep_api.h"
+#include "cmsis.h"
+#include "mbed_interface.h"
+
+#if defined(RDA_SLEEP_MODE)
+#include "rda_sleep_api.h"
+#endif /* RDA_SLEEP_MODE */
+
+void hal_sleep(void)
+{
+#if (DEVICE_SEMIHOST == 1)
+    // ensure debug is disconnected
+    mbed_interface_disconnect();
+#endif
+
+#if defined(RDA_SLEEP_MODE)
```

Comment:
what does `RDA_SLEEP_MODE` mean - is it defined? Where is it documented?

Reply:
User can decide whether to allow sleep and set wakeup source (GPIO or UART) if RDA_SLEEP_MODE defined. We don't have public technical reference manual for now.